### PR TITLE
Add codemods image

### PR DESCRIPTION
**Pull request by MaggieAppleton**
I popped in the quickie codemods illustration to Laurie's article.

However, it's an inline image that's stored in Cloudinary.
Couldn't figure out what our system is for images on articles – eg. where to store the image. How to pass it into the article template. How to have it automagically become the OG Image for social media shares.
Didn't want to mess anything up - @vojtaholik I know you're well versed in best practices and techniques for image handling on posts. What's the right approach here?

Screenshot 2020-12-08 at 21 25 34

### CODE REVIEW

**Comment by joelhooks** [comment]
I think we are _probably_ better off using `next/image` and letting them handle the cdn/optimisation stuff in most circumstances.

Also curious how we want to handle share cards. I'd love for them to have individual flavor, but also want to respect the volume and our capacity to create stuff.

Probably a "sensible default" and then custom overrides?

File: src/layouts/ultimate-guide.tsx [hunk 1]
```diff
@@ -32,17 +34,35 @@ const UltimateGuideLayout: FunctionComponent<LayoutProps> = ({
...
+            {coverImage && (
+              <div className="mt-4">
+                <Image
+                  src={coverImage.url}
+                  alt={coverImage.alt || title}
+                  width={1280}
+                  height={720}
+                />
+              </div>
+            )}
```
Comment (vojtaholik) [inline src/layouts/ultimate-guide.tsx:60]: using standard 16:9 ratio
can optionally specify `alt` attribute

File: src/pages/learn/javascript/codemods-with-babel-plugins/index.mdx [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
   contributors: ['Laurie Barth'],
+  coverImage: {
+    url:
+      'https://res.cloudinary.com/dg3gyk0gu/image/upload/v1607523756/egghead-next-ebombs/article-illustrations/codemods.png',
+    alt:
+      'abstract illustration of sphere morphing into ring, stop motion style',
+  },
+  ogImage: {
+    url: `https://og-image-react-egghead.now.sh/article/Codemods%20with%20Babel%20Plugins?bgImage=https://res.cloudinary.com/dg3gyk0gu/image/upload/v1607527530/egghead-next-ebombs/article-illustrations/card.png&author=Laurie%20Barth`,
+  },
```
Comment (vojtaholik) [inline src/pages/learn/javascript/codemods-with-babel-plugins/index.mdx:12]: Using card generator with three variables: (`title`, `bgImage`, `author`)
```jsx
`https://og-image-react-egghead.now.sh/article/${title}?bgImage=${bgImage}&author=${author}`
```
`bgImage` points to: 
<a href="https://res.cloudinary.com/dg3gyk0gu/image/upload/v1607527530/egghead-next-ebombs/article-illustrations/card.png"></a>
which combined with [article template](https://github.com/eggheadio/og-image-egghead/blob/master/src/_lib/template.js#L192) results in: 
<a href="https://og-image-react-egghead.now.sh/article/Codemods%20with%20Babel%20Plugins?bgImage=https://res.cloudinary.com/dg3gyk0gu/image/upload/v1607527530/egghead-next-ebombs/article-illustrations/card.png&author=Laurie%20Barth"></a>

**Great thing about this is that if we already have a `coverImage`, we can utilize it for card's background.** We can also still pass a custom made shard card.

Here's a [template in Figma](https://www.figma.com/file/698PNWT1OjHImlj1oryw78/og-images?node-id=0%3A1)

File: src/pages/learn/javascript/codemods-with-babel-plugins/index.mdx [hunk 3]
```diff
@@ -1,6 +1,15 @@
 export const meta = {
   title: `Codemods with Babel Plugins`,
   contributors: ['Laurie Barth'],
+  coverImage: {
+    url:
+      'https://res.cloudinary.com/dg3gyk0gu/image/upload/v1607523756/egghead-next-ebombs/article-illustrations/codemods.png',
+    alt:
+      'abstract illustration of sphere morphing into ring, stop motion style',
+  },
```
Comment (vojtaholik) [inline src/pages/learn/javascript/codemods-with-babel-plugins/index.mdx:9]: didnt forget to compress the image before uploading to cloudinary (💙 [alfred tinypng workflow](https://www.alfredapp.com/blog/productivity/featured-workflow-compress-images-with-tinypng/))

**Comment by joelhooks** [comment]
cloudinary is fine for images but i think we need to consider the file names a bit more, it’s marginal but they matter for both accessibility and seo. 

in this case the image is strictly decorative, so not a huge concern, but the file name could describe it still beyond “card”


was also wondering if the card generator couldn’t be “local” to this app instead of another app and use shared images via next/image 🤔


that’s definitely out of scope though 😅

File: src/layouts/ultimate-guide.tsx [hunk 4]
```diff
@@ -28,21 +34,47 @@ const UltimateGuideLayout: FunctionComponent<LayoutProps> = ({
           title,
           description,
           url,
-          images: ogImage ? [ogImage] : undefined,
+          images: ogImage
+            ? [ogImage]
+            : [
+                {
+                  url: `https://og-image-react-egghead.now.sh/article/Codemods%20with%20Babel%20Plugins?author=${
```
Comment (vojtaholik) [inline src/layouts/ultimate-guide.tsx:41]: default share card looks like this: <a href="https://og-image-react-egghead.now.sh/article/Codemods%20with%20Babel%20Plugins?author=Laurie%20Barth"></a>

**Comment by vojtaholik** [comment]
@joelhooks can provide `alt` attr to both `coverImage` and `ogImage`. I'm concerned about file structure in cloudinary, think it should mirror the app?

**Comment by joelhooks** [comment]
the alt is important too, but the actual filename of the cover image is what i was talking about

the path/url is a minor/marginal thing, but still relevant and the closer to the content we can make that without being a total PITA the better.

**Review by vojtaholik — APPROVED** [review]

File: src/pages/learn/javascript/codemods-with-babel-plugins/index.mdx (hunk 2 above)
Comment (MaggieAppleton) [reply src/pages/learn/javascript/codemods-with-babel-plugins/index.mdx:12]: Looks great 👍

**Pull request merged**